pkg/apps/registry/deployconfig/etcd/etcd.go

```diff
@@ -72,12 +78,21 @@ type ScaleREST struct {
 	store *registry.Store
 }
 
-// ScaleREST implements Patcher
 var _ = rest.Patcher(&ScaleREST{})
+var _ = rest.GroupVersionKindProvider(&ScaleREST{})
 
 // New creates a new Scale object
 func (r *ScaleREST) New() runtime.Object {
-	return &extensions.Scale{}
+	return &autoscaling.Scale{}
+}
+
+func (r *ScaleREST) GroupVersionKind(containingGV schema.GroupVersion) schema.GroupVersionKind {
+	switch containingGV {
```

Comment:
@liggitt explicitly set us to use `extensions/v1beta1.Scale` only, not sure if this change is desired.

Reply:
long term we want to move to autoscaling/Scale

Reply:
but worth checking with @liggitt

Reply:
> @liggitt explicitly set us to use extensions/v1beta1.Scale only, not sure if this change is desired.

We need to make sure that any future version uses autoscaling. This code does that.

Reply:
@soltysh this is not changing the default, if I do `oc scale dc/foo` I'm still send extensions/Scale and get extensions/Scale. I believe this is just so we have the **right** internal type.

Reply:
Roger that.